src/frontend/sharedTypes/navigation.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  CompositeScreenProps,
+  NavigatorScreenParams,
+} from '@react-navigation/native';
+import type {RoleId, RoleIdForNewInvite} from '@mapeo/core/dist/roles';
+import {NativeStackScreenProps} from '@react-navigation/native-stack';
+import {MessageDescriptor} from 'react-intl';
+import {BottomTabScreenProps} from '@react-navigation/bottom-tabs';
+
+export interface TabBarIconProps {
+  size: number;
+  focused: boolean;
+  color: string;
+}
+
+export type DeviceType = 'mobile' | 'desktop';
+
+export type DeviceRole = RoleId;
+
+export type DeviceRoleForNewInvite = RoleIdForNewInvite;
+
+export type InviteProps = {
+  name: string;
+  deviceType: DeviceType;
+  deviceId: string;
+  role: DeviceRoleForNewInvite;
+};
+
+export type HomeTabsList = {
+  Map: undefined;
+  Camera: undefined;
+  Tracking: undefined;
+  ObservationsList: undefined;
+};
+
+export type TabName = keyof HomeTabsList;
+
+export type AppList = {
```

Comment:
Nit: It took me a while to figure out what this is. I think a name like `RootStackParamList` would be clearer, and then calling `HomeTabsList` -> `HomeTabsParamList` and `DeviceNamingScreens` -> `DeviceNamingParamsList` would be clearer, as I understand that these are defining the prop types for each screen. A code comment would be helpful for a future maintainer.